src/dfmt/formatter.d

```diff
@@ -1053,6 +1053,14 @@ private:
             if (niBraceDepth > 0)
                 niBraceDepth--;
 
+            // Account for possible function literals in this array which offset
+            // the previously set index (pos). Fixes issue #432.
+            while(astInformation.indentInfoSortedByEndLocation[pos].endLocation !=
```

Comment:
```suggestion
            while(astInformation.indentInfoSortedByEndLocation[pos].endLocation <
```

avoids endless loop / array out of bounds in case AST info isn't fully accurate, e.g. in broken AST. You might want to check for it to be equal afterwards to avoid doing any indents in the broken case at all.

Reply:
I have implemented that change in the latest commit, but I don't think that should be pulled yet since I believe there can be a problem, see below